src/ekat/logging/ekat_log_mpi.hpp

```diff
@@ -15,35 +15,20 @@
 // methods.
 namespace ekat {
 
-/* Console log ignores MPI Rank; all ranks will write to the console without rank ids.
-  File logs (if used) will still have their rank id in their filenames.
-*/
-struct LogIgnoreRank {
-  // leave sinks alone
-  static void update_sinks(std::vector<spdlog::sink_ptr>& sinks) {}
-
-  // leave name alone
-  static std::string name_with_rank(const std::string& name) {
-    return name;
-  }
-};
-
-
 /* Suppress console output from ranks > 0.  All ranks still write to files, if
   file logging is enabled.
 */
 struct LogOnlyRank0 {
-  // erase the console sink for all ranks != 0
-  static void update_sinks(std::vector<spdlog::sink_ptr>& sinks) {
-    const Comm mpicomm(MPI_COMM_WORLD);
+  // nullify the console sink for all ranks != 0
+  static void update_sinks(std::vector<spdlog::sink_ptr>& sinks, const ekat::Comm& mpicomm=ekat::Comm()) {
```

Comment:
Related to the above: if we only deal with one sink, I think it'd be best to pass a single sink ptr to this routine, and let the Logger class distinguish the concept of file and console sinks.